pkg/config/remote/client.go

```diff
@@ -0,0 +1,201 @@
+		return nil, errors.Wrap(err, "could not acquire agent auth token")
+	}
+	ctx, close := context.WithCancel(context.Background())
+	md := metadata.MD{
+		"authorization": []string{fmt.Sprintf("Bearer %s", token)},
+	}
+	ctx = metadata.NewOutgoingContext(ctx, md)
+	grpcClient, err := agentgrpc.GetDDAgentSecureClient(ctx, dialOpts...)
+	if err != nil {
+		close()
+		return nil, err
+	}
+	partialClient, err := uptane.NewPartialClient()
+	if err != nil {
+		close()
+		return nil, err
+	}
+	enabledProducts := make(map[data.Product]struct{})
+	for _, product := range products {
+		enabledProducts[product] = struct{}{}
+	}
+	return &Client{
+		ctx:                ctx,
+		facts:              facts,
+		enabledProducts:    enabledProducts,
+		grpc:               grpcClient,
+		close:              close,
+		pollInterval:       1 * time.Second,
+		partialClient:      partialClient,
+		apmSamplingUpdates: make(chan APMSamplingUpdate, 8),
+		configs:            newConfigs(),
+	}, nil
+}
+
+// Close closes the client
+func (c *Client) Close() {
+	c.close()
+}
+
+func (c *Client) pollLoop() {
+	for {
+		select {
+		case <-c.ctx.Done():
+			return
+		case <-time.After(c.pollInterval):
+			c.lastPollErr = c.poll()
+			if c.lastPollErr != nil {
+				log.Errorf("could not poll remote-config agent service: %v", c.lastPollErr)
+			}
+		}
+	}
+}
+
+func (c *Client) products() []data.Product {
+	var products []data.Product
+	for product := range c.enabledProducts {
+		products = append(products, product)
+	}
+	return products
+}
+
+func (c *Client) poll() error {
+	c.m.Lock()
+	defer c.m.Unlock()
+	state := c.partialClient.State()
+	lastPollErr := ""
+	if c.lastPollErr != nil {
+		lastPollErr = c.lastPollErr.Error()
+	}
+	response, err := c.grpc.ClientGetConfigs(c.ctx, &pbgo.ClientGetConfigsRequest{
+		Client: &pbgo.Client{
+			Id:      c.facts.ID,
+			Name:    c.facts.Name,
+			Version: c.facts.Version,
+			State: &pbgo.ClientState{
+				RootVersion:    state.RootVersion,
+				TargetsVersion: state.TargetsVersion,
+				Configs:        c.configs.state(),
+				HasError:       c.lastPollErr != nil,
+				Error:          lastPollErr,
+			},
+			Products: data.ProductListToString(c.products()),
+		},
+	})
+	if err != nil {
+		return err
+	}
+	err = c.partialClient.Update(response)
+	if err != nil {
+		return err
+	}
+	configFiles, err := c.buildConfigFiles()
+	if err != nil {
+		return err
+	}
+	updates := c.configs.update(c.products(), configFiles)
+	c.publishUpdates(updates)
+	return nil
+}
+
+func (c *Client) buildConfigFiles() (configFiles, error) {
+	targets, err := c.partialClient.Targets()
+	if err != nil {
+		return nil, err
+	}
+	var configFiles configFiles
+	for targetPath, target := range targets {
+		targetPathMeta, err := data.ParseFilePathMeta(targetPath)
+		if err != nil {
+			return nil, err
+		}
+		if _, productEnabled := c.enabledProducts[targetPathMeta.Product]; productEnabled {
+			targetContent, err := c.partialClient.TargetFile(targetPath)
```

Comment:
I think it would be clearer if the partial client returned a list of verified configs. All errors should already be caught by partialClient.Update(response)

verifiedConfigs := c.partialClient.GetVerifiedConfigs() 
// trim products